lib/benchee/benchmark.ex

```diff
@@ -43,9 +43,23 @@ defmodule Benchee.Benchmark do
   @spec measure(Suite.t, module) :: Suite.t
   def measure(suite = %Suite{jobs: jobs, configuration: config}, printer \\ Printer) do
     printer.configuration_information(suite)
-    run_times = record_runtimes(jobs, config, printer)
-
-    %Suite{suite | run_times: run_times}
+    results = record_results(jobs, config, printer)
+    run_times = parse_results(results, fn({key, {run_times, _}}) ->
+      {key, run_times}
+    end)
+    memory_usages = parse_results(results, fn({key, {_, memory_usages}}) ->
+      {key, memory_usages}
+    end)
+
+    %Suite{suite | run_times: run_times, memory_usages: memory_usages}
+  end
+
+  defp parse_results(results, func) do
+    results
+      |> Enum.map(fn({input, map}) ->
+        {input, map |> Enum.map(func) |> Map.new}
+      end)
+      |> Map.new
```

Comment:
more a fan of 

```
results
|> ...
|> ...
```

indentation wise ofer here, plus not totally wure but for mapping stuff/anon functions I mostly prefer 

```
start
|> Enum.map ,,, fn ->
   end
|> 
```

(not breaking the pipe)

but that's all style and my fault for not having a linter configured :)